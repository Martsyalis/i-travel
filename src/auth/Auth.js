import React, { PureComponent } from 'react';
import { Switch, Route, Link, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { signin, signup } from './actions';
import styled from 'styled-components';
import Credentials from './Credentials';

class Auth extends PureComponent {

  render(){
    const redirect = this.props.location.state ? this.props.location.state.from : '/';

    if(this.props.user) return <Redirect to={redirect}/>;


    return (
      <CenteredDiv>
        <h1>Welcome!</h1>
        <h5>
          This is iTravel
          <Link to="/auth/signin" > sign In Here </Link> if You already have an account with us or<Link to="/auth/signup" > signUp here </Link> to make a new one
        </h5>
        <Switch>
          <Route path="/auth/signin" component={() => (
            <Credentials action="Sign In" submit={this.props.signin}/>
          )}/>
          <Route path="/auth/signup" render={() => (
            <Credentials action="Sign Up" submit={this.props.signup} allowName={true}/>
          )}/>
        </Switch>
        {this.props.error && <Error>{ this.props.error }</Error>}
      </CenteredDiv>
    );
  }
}

export default withRouter(connect(
  ({ auth }) => ({ 
    error: auth.error,
    user: auth.user 
  }),
  { signup, signin }
)(Auth));

const CenteredDiv = styled.div`
text-align: center;
`;

const Error = styled.pre`
color: red;
`;