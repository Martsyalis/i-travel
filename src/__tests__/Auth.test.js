import React from 'react';

import { shallow } from 'enzyme';

import Auth from '../auth/Auth';

describe('Auth', () => {
  it('shows the Auth component', () => {
    const wrapper = shallow(<Auth />);
    expect(wrapper).toMatchSnapshot();
  });
});
