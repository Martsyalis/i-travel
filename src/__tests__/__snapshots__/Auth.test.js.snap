// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth shows the Auth component 1`] = `
<Route
  render={[Function]}
/>
`;
